### ISSUE TITLE
Request: unify authentication implementation so they can be combined

### ISSUE THREAD

**Issue by kbrandwijk** [graphcool-examples/functions#48]
It would be great if it was easy to combine multiple custom authentication methods into one project. For this, it's necessary to unify the implementations, to make sure none of them use the same query or mutation names, or the same fields in the User Type, etc.

**Comment by marktani** [comment]
I don't think we should expect different examples to be compatible to each other with no changes. 

Instead, I'd like to see a new example that shows how to combine two authentication methods and offers additional functionality (like allowing to authenticate via email/password as well as Facebook for the same user if the email address matches).

**Comment by kbrandwijk** [comment]
No not without changes. The idea was that we might come up with a sort of 'template' for implementing an authentication provider, so that it's easier to combine them.
